cmd/promtool/main.go

```diff
@@ -377,6 +384,18 @@ func CheckMetrics() int {
 	return 0
 }
 
+// CheckExpression validates if the input expression is a valid PromQL expression
+func CheckExpression(expr string) int {
```

Comment:
I know we would break consistency, but can we just return error here? This function is super basic. This will allow us to unit test it without this 
```
defer func() { os.Stderr = stderr }()
os.Stderr = os.NewFile(0, os.DevNull)
```

But I guess this would mean changing interface for all commands here..

Reply:
I could have a go at it at some point later, to refactor all of it to return an error instead of just an integer! I can report an issue to track it if that helps :). Doing it in the same PR would either make this one a lot bigger (if changing all of them) or introduce inconsistency (if just changing this single one) :sweat_smile: .